File: examples/simple-chatbot/client/ios/SimpleChatbot/views/settings/SettingsManager.swift
import Foundation

class SettingsManager {
    private static let preferencesKey = "settingsPreference"

    static func getSettings() -> SettingsPreference {
        if let data = UserDefaults.standard.data(forKey: preferencesKey),
           let settings = try? JSONDecoder().decode(SettingsPreference.self, from: data) {
            return settings
        } else {
            // default values in case we don't have any settings
            return SettingsPreference(enableMic: true, backendURL: "http://YOUR_IP:7860")
        }
    }
    
    static func updateSettings(settings: SettingsPreference) {
        if let data = try? JSONEncoder().encode(settings) {
            UserDefaults.standard.set(data, forKey: preferencesKey)
        }
    }
}

File: examples/simple-chatbot/client/react-native/src/settings/SettingsManager.ts
import AsyncStorage from '@react-native-async-storage/async-storage';

export interface SettingsManager {
  enableCam: boolean;
  enableMic: boolean;
  backendURL: string;
}

// Define the settings object
const defaultSettings: SettingsManager = {
  enableCam: false,
  enableMic: true,
  backendURL: process.env.EXPO_SIMPLE_CHATBOT_SERVER || "",
};

export class SettingsManager {
  private static preferencesKey = 'settingsPreference';

  static async getSettings(): Promise<SettingsManager> {
    try {
      const data = await AsyncStorage.getItem(this.preferencesKey);
      if (data !== null) {
        return JSON.parse(data) as SettingsManager;
      } else {
        return defaultSettings;
      }
    } catch (error) {
      console.error("Failed to load settings:", error);
      return defaultSettings;
    }
  }

  static async updateSettings(settings: SettingsManager): Promise<void> {
    try {
      const data = JSON.stringify(settings);
      await AsyncStorage.setItem(this.preferencesKey, data);
    } catch (error) {
      console.error("Failed to save settings:", error);
    }
  }
}

